app/scripts/modules/core/src/pipeline/config/stages/bakeManifest/bakeManifestConfig.controller.ts

```diff
@@ -32,6 +40,26 @@ export class BakeManifestConfigCtrl implements IController {
     this.expectedArtifacts = expectedArtifactService.getExpectedArtifactsAvailableToStage(this.$scope.stage, this.$scope.$parent.pipeline);
   }
 
+  public addInputArtifact() {
+    if (!this.$scope.stage.inputArtifacts) {
+      this.$scope.stage.inputArtifacts = [ BakeManifestConfigCtrl.defaultInputArtifact() ];
+    }
+
+    this.$scope.stage.inputArtifacts.push(BakeManifestConfigCtrl.defaultInputArtifact())
```

Comment:
You're adding the default artifact twice if you enter the `if` block

Reply:
That's the intent - the first value is "special" and the command doesn't make sense w/o it.

Reply:
Ah, this had confused me---I think this probably relates to my comment on https://github.com/spinnaker/rosco/pull/251#event-1589491142 where it might lead to confusing code/bugs if we have an array where the first value is special rather than pulling that into its own variable.  Maybe even a comment to clarify this would help?